test/jdk/java/awt/Robot/ScreenCaptureRobotTest.java

```diff
@@ -0,0 +1,147 @@
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.BorderLayout;
+import java.awt.Canvas;
+import java.awt.Color;
+import java.awt.EventQueue;
+import java.awt.Font;
+import java.awt.Frame;
+import java.awt.Graphics;
+import java.awt.GraphicsEnvironment;
+import java.awt.Point;
+import java.awt.Rectangle;
+import java.awt.Robot;
+import java.awt.image.BufferedImage;
+
+/*
+ * @test
+ * @key headful
+ * @bug 8342098
+ * @summary Verify that the image captured from the screen using a Robot
+ * and the source image are same.
+ * @run main ScreenCaptureRobotTest
+ */
+public class ScreenCaptureRobotTest {
+
+    private static int delay = 500;
+    private static Frame frame;
+    private static volatile Canvas canvas;
+    private static BufferedImage realImage;
+    private static BufferedImage displayImage;
+
+    public static void main(String[] args) throws Exception {
+        try {
+            EventQueue.invokeAndWait(ScreenCaptureRobotTest::initializeGUI);
+            doTest();
+        } finally {
+            EventQueue.invokeAndWait(ScreenCaptureRobotTest::disposeFrame);
+        }
+    }
+
+    private static void initializeGUI() {
+        frame = new Frame("ScreenCaptureRobotTest Frame");
+        realImage = GraphicsEnvironment.getLocalGraphicsEnvironment()
+                .getDefaultScreenDevice().getDefaultConfiguration()
+                .createCompatibleImage(200, 100);
+
+        Graphics g = realImage.createGraphics();
+        g.setColor(Color.yellow);
+        g.fillRect(0, 0, 200, 100);
+        g.setColor(Color.red);
+        g.setFont(new Font("SansSerif", Font.BOLD, 20));
+        g.drawString("Capture This", 10, 40);
+        g.dispose();
+        displayImage = realImage;
+
+        canvas = new ImageCanvas();
+        canvas.setBackground(Color.YELLOW);
+        frame.setLayout(new BorderLayout());
+        frame.add(canvas);
+        frame.setSize(300, 200);
+        frame.setLocation(100, 100);
+        frame.setVisible(true);
+        canvas.requestFocus();
+    }
+
+    private static void doTest() throws Exception {
+        Robot robot;
+        robot = new Robot();
+        robot.delay(delay);
+        robot.waitForIdle();
+
+        Point pnt = canvas.getLocationOnScreen();
+        Rectangle rect = new Rectangle(pnt.x + 10, pnt.y + 10, 200, 100);
+
+        // Capturing Image using Robot
+        BufferedImage capturedImage = robot.createScreenCapture(rect);
+
+        if (!compareImages(capturedImage, realImage)) {
+            String errorMessage = "FAIL : Captured Image is different from "
+                    + "the actual image";
+            System.err.println("Test failed");
+            throw new RuntimeException(errorMessage);
```

Comment:
```suggestion
            System.err.println("Test failed");
            saveImage(capturedImage,"robotCapture.png");
            saveImage(realImage,"realImage.png");
            throw new RuntimeException(errorMessage);
```

Reply:
@manukumarvs 
Right, test passed with sRGB profile
in MacOS 14 & earlier version 12 as well.